src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,70 @@
+package lotto.controller;
+
+import lotto.dto.BonusLottoNumber;
+import lotto.dto.BuyLottos;
+import lotto.dto.LottoPurchaseCost;
+import lotto.dto.MatchLottoResult;
+import lotto.dto.RateOfReturn;
+import lotto.dto.WinningLottoNumbers;
+import lotto.service.LottoService;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final LottoService lottoService;
+
+    public LottoController(InputView inputView, OutputView outputView, LottoService lottoService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.lottoService = lottoService;
+    }
+
+    public void run() {
+        LottoPurchaseCost lottoPurchaseCost = inputLottoPurchaseCost();
+        BuyLottos buyLottos = lottoService.buyLottos(lottoPurchaseCost);
+        outputView.printLottos(buyLottos.numbers());
+
+        WinningLottoNumbers winningLottoNumbers = getWinningLottoNumbers();
+        BonusLottoNumber bonusLottoNumber = getBonusLottoNumber(winningLottoNumbers);
+
+        MatchLottoResult matchLottoResult = lottoService.matchLottoNumber(buyLottos, winningLottoNumbers,
+                bonusLottoNumber);
+        RateOfReturn rateOfReturn = lottoService.calcRateOfReturn(lottoPurchaseCost, matchLottoResult);
+        outputView.printResult(matchLottoResult, rateOfReturn);
+    }
+
+    private LottoPurchaseCost inputLottoPurchaseCost() {
+        while (true) {
+            try {
+                outputView.printLottoPurchaseCostMessage();
+                return LottoPurchaseCost.of(inputView.InputLottoPurchaseAmount());
+            } catch (IllegalArgumentException e) {
+                outputView.printErrorMessage(e);
+            }
+        }
+    }
+
+    private WinningLottoNumbers getWinningLottoNumbers() {
+        while (true) {
+            try {
+                outputView.printInputLottoWinningNumbersMessage();
+                return WinningLottoNumbers.of(inputView.InputLottoWinningNumbers());
+            } catch (IllegalArgumentException e) {
+                outputView.printErrorMessage(e);
```

Comment:
```suggestion
            } catch (IllegalArgumentException e) {
                outputView.printErrorMessage(e.getMessage());
```
이런식으로 `String message`를 넘기는건 어떤가요?

Reply:
동의합니다 !! 지금 보니, 예외 자체를 뷰에 넘기고 있었네요..
반영하겠습니다. 피드백 감사합니다 👍